compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -1107,12 +1380,117 @@ impl<'a> PartialEvaluator<'a> {
     }
 
     fn try_eval_stmt(&mut self, stmt_id: StmtId) -> Result<EvalControlFlow, Error> {
-        // If the statement is classical, we can just evaluate it.
-        if self.is_classical_stmt(stmt_id) {
-            self.eval_classical_stmt(stmt_id)
+        let stmt = self.get_stmt(stmt_id);
+        match stmt.kind {
+            StmtKind::Expr(expr_id) => self.try_eval_expr(expr_id),
```

Comment:
Why doesn't handling of `StmtKind::Expr` have to handle control flow adjustments? Please add a comment.

Reply:
Comment added.